### PR TITLE
net/socket: move si_send/recv into sendmsg/recvmsg

### PR DESCRIPTION
**Pull request by PeterBee97**
## Summary
Implement si_send/sendto/recvfrom with si_sendmsg/recvmsg, instead of
the other way round. 
Multi-iov cases has been handled in Unix domain sockets and inet sockets.

### CODE REVIEW

File: net/local/local_sendmsg.c [hunk 1]
```diff
@@ -162,7 +270,46 @@ ssize_t psock_local_sendto(FAR struct socket *psock,
   /* Release our reference to the half duplex FIFO */
 
   local_release_halfduplex(conn);
+
+#else
+  nsent = -EISCONN;
```
Comment (xiaoxiang781216) [inline net/local/local_sendmsg.c:275]:
```suggestion
  return -EISCONN;
```

File: net/pkt/pkt_sockif.c [hunk 2]
```diff
@@ -341,7 +328,7 @@ static int pkt_bind(FAR struct socket *psock,
 
   if (addr->sa_family != AF_PACKET || addrlen < sizeof(struct sockaddr_ll))
     {
-      nerr("ERROR: Invalid address length: %d < %d\n",
+      nerr("ERROR: Invalid address length: %d < %u\n",
```
[the remark this answers is not in the document]
Comment (PeterBee97) [reply net/pkt/pkt_sockif.c]: zu?

[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply net/pkt/pkt_sockif.c]: Yes, you are right.

File: net/inet/inet_sockif.c [hunk 3]
```diff
@@ -1224,6 +1222,68 @@ static ssize_t inet_sendto(FAR struct socket *psock, FAR const void *buf,
...
+ *   On success, returns the number of characters sent.  On  error, a negated
+ *   errno value is returned (see sendmsg() for the list of appropriate error
+ *   values.
+ *
+ ****************************************************************************/
+
+static ssize_t inet_sendmsg(FAR struct socket *psock,
+                            FAR struct msghdr *msg, int flags)
+{
+  FAR void *buf = msg->msg_iov->iov_base;
```
[the remark this answers is not in the document]
Comment (PeterBee97) [reply net/inet/inet_sockif.c:1247]: The buf pointer is reused in the temporary support for multi-iov memcpy below, should we use another pointer or keep it as-is for now?

[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply net/inet/inet_sockif.c:1247]: Ok, let's keep it as before.

**Review by xiaoxiang781216 — APPROVED** [review]
LGTM.

**Pull request merged**